src/PKSim.Core/Services/SimulationConfigurationTask.cs

```diff
@@ -125,5 +131,21 @@ public SimulationConfiguration CreateFor(Simulation simulation, bool shouldValid
 
          return simulationConfiguration;
       }
+
+      private void createInitialConditionsForDefinedProteins(Simulation simulation, SimulationConfiguration simulationConfiguration, Module module, IReadOnlyList<MoleculeBuilder> allDefinedProteins)
+      {
+         simulationConfiguration.ExpressionProfiles.Each(expressionProfile => addInitialConditions(expressionProfile, module.SpatialStructure, allDefinedProteins, simulation));
+      }
+
+      private InitialConditionsBuildingBlock createInitialConditionsForCompounds(Module module, Simulation simulation, IReadOnlyList<MoleculeBuilder> compoundMolecules)
+      {
+         return _initialConditionsCreator.CreateFor(module.SpatialStructure, compoundMolecules, simulation);
+      }
+
+      private void addInitialConditions(ExpressionProfileBuildingBlock expressionProfile, SpatialStructure spatialStructure, IReadOnlyList<MoleculeBuilder> molecules, Simulation simulation)
+      {
+         var initialConditions = _initialConditionsCreator.CreateFor(spatialStructure, molecules.Where(molecule => Equals(molecule.Name, expressionProfile.MoleculeName)).ToList(), simulation);
```

Comment:
FindByName? 
Feels like we need an overload with only one molecule in _initialConditionsCreator .CreateFor